packages/devkit/src/utils/update-package-scripts.ts

```diff
@@ -0,0 +1,224 @@
+import type { NxJsonConfiguration } from 'nx/src/config/nx-json';
+import type { Tree } from 'nx/src/generators/tree';
+import type {
+  CreateNodes,
+  CreateNodesFunction,
+  CreateNodesResult,
+} from 'nx/src/utils/nx-plugin';
+import type { PackageJson } from 'nx/src/utils/package-json';
+import { dirname } from 'path';
+import * as yargs from 'yargs-parser';
+import { requireNx } from '../../nx';
+
+const { glob, readJson, readNxJson, workspaceRoot, writeJson } = requireNx();
+
+type TargetCommand = {
+  command: string;
+  target: string;
+  configuration?: string;
+};
+
+export async function updatePackageScripts(
+  tree: Tree,
+  createNodesTuple: CreateNodes
+): Promise<void> {
+  const nxJson = readNxJson(tree);
+
+  const [pattern, createNodes] = createNodesTuple;
+  const files = glob(tree, [pattern]);
+
+  for (const file of files) {
+    const projectRoot = dirname(file);
+    await processProject(tree, projectRoot, file, createNodes, nxJson);
+  }
+}
+
+async function processProject(
+  tree: Tree,
+  projectRoot: string,
+  projectConfigurationFile: string,
+  createNodesFunction: CreateNodesFunction,
+  nxJsonConfiguration: NxJsonConfiguration
+) {
+  const packageJsonPath = `${projectRoot}/package.json`;
+  if (!tree.exists(packageJsonPath)) {
+    return;
+  }
+  const packageJson = readJson<PackageJson>(tree, packageJsonPath);
+  if (!packageJson.scripts || !Object.keys(packageJson.scripts).length) {
+    return;
+  }
+
+  const result = await createNodesFunction(
+    projectConfigurationFile,
+    {},
+    { nxJsonConfiguration, workspaceRoot }
+  );
+
+  const targetCommands = getInferredTargetCommands(result);
+  if (!targetCommands.length) {
+    return;
+  }
+
+  // exclude package.json scripts from nx
+  packageJson.nx ??= {};
+  packageJson.nx.includedScripts ??= [];
```

Comment:
@FrozenPandaz @jaysoo do we want to include all scripts by default and only remove the ones we end up replacing?

Right now, the PR excludes everything if there is no `nx` entry in the `package.json`. If there's an `nx` entry, it's respected, but replaced scripts are removed.